src/TSMapEditor/UI/Windows/ScriptsWindow.cs

```diff
@@ -515,15 +581,49 @@ private void ListScripts()
         {
             lbScriptTypes.Clear();
 
-            foreach (var script in map.Scripts)
+            List<Script> sortedScripts = map.Scripts;
+
+            var shouldViewTop = false; // when filtering the scroll bar should update so we use a flag here
+            if (tbFilter.Text != string.Empty && tbFilter.Text != tbFilter.Suggestion)
             {
-                lbScriptTypes.AddItem(new XNAListBoxItem() { Text = script.Name, Tag = script });
+                sortedScripts = sortedScripts.Where(sortedScript => sortedScript.Name.Contains(tbFilter.Text, StringComparison.CurrentCultureIgnoreCase)).ToList();
+                shouldViewTop = true;
             }
+
+            switch (ScriptSortMode)
+            {
+                case ScriptSortMode.Color:
+                    sortedScripts = sortedScripts.OrderBy(script => script.EditorColor).ThenBy(script => script.ININame).ToList();
+                    break;
+                case ScriptSortMode.Name:
+                    sortedScripts = sortedScripts.OrderBy(script => script.Name).ThenBy(script => script.ININame).ToList();
+                    break;
+                case ScriptSortMode.ColorThenName:
+                    sortedScripts = sortedScripts.OrderBy(script => script.EditorColor).ThenBy(script => script.Name).ToList();
+                    break;
+                case ScriptSortMode.ID:
+                default:
+                    sortedScripts = sortedScripts.OrderBy(script => script.ININame).ToList();
+                    break;
```

Comment:
No need to turn it into a list.

```suggestion
                case ScriptSortMode.Color:
                    sortedScripts = sortedScripts.OrderBy(script => script.EditorColor).ThenBy(script => script.ININame);
                    break;
                case ScriptSortMode.Name:
                    sortedScripts = sortedScripts.OrderBy(script => script.Name).ThenBy(script => script.ININame);
                    break;
                case ScriptSortMode.ColorThenName:
                    sortedScripts = sortedScripts.OrderBy(script => script.EditorColor).ThenBy(script => script.Name);
                    break;
                case ScriptSortMode.ID:
                default:
                    sortedScripts = sortedScripts.OrderBy(script => script.ININame);
                    break;
```